### ISSUE TITLE
OneArmedIf + EmptyDefine

### ISSUE THREAD

**Issue by jcubic** [schemedoc/surveys#10]
It seems this is missing. In Guile and Chicken.
```
(define x)
(eq? (if #f #f) x)
#t
```
so define without value may be another way to define unspecified value.
So those two services can be somehow connected or linked.

**Comment by lassik** [comment]
Indeed. We could merge these surveys (and maybe more) into one survey about what happens when different expressions (`if`, `cond`, `define`, `begin`, ...) have a missing body.

**Comment by lassik** [comment]
In some implementations, an empty `(begin)` returns a different value depending on the situation. (Gauche used to return the last value in a VM register.)

**Comment by lassik** [comment]
Some expressions to try:

```Scheme
(define foo)
(define (foo))
(begin)
(cond)
(cond (else))
(if 1 2)
(guard (e (else)) (error "foo"))
```
